internal/collector/config.go

```diff
@@ -14,27 +14,30 @@ import (
 )
 
 // ReconcileCollectorConfig reconciles a collector config specifically for the collector defined by the factory
-func (f *Factory) ReconcileCollectorConfig(er record.EventRecorder, k8sClient client.Client, namespace, name, collectorConfig string, owner metav1.OwnerReference) error {
+func (f *Factory) ReconcileCollectorConfig(er record.EventRecorder, k8sClient client.Client, instance *logging.ClusterLogging, name, collectorConfig string, owner metav1.OwnerReference) error {
 	log.V(3).Info("Updating ConfigMap and Secrets")
 	if f.CollectorType == logging.LogCollectionTypeFluentd {
 		collectorConfigMap := runtime.NewConfigMap(
-			namespace,
+			instance.Namespace,
 			name,
 			map[string]string{
 				"fluent.conf":         collectorConfig,
 				"run.sh":              fluentd.RunScript,
 				"cleanInValidJson.rb": fluentd.CleanInValidJson,
 			},
 		)
+		utils.SetCommonLabels(collectorConfigMap, instance, constants.CollectorName)
```

Comment:
I think there may be a more clever way to add these labels by modifying the [runtime.Initialize](https://github.com/openshift/cluster-logging-operator/blob/master/internal/runtime/runtime.go#L90)

```
func Initialize(o runtime.Object, namespace, name string, visitors ... func(meta metav1.Object)  {
	m := Meta(o)
	m.SetNamespace(namespace)
	m.SetName(name)
	o.GetObjectKind().SetGroupVersionKind(GroupVersionKind(o))
	m.SetLabels(map[string]string{
		"pod-security.kubernetes.io/enforce":             "privileged",
		"security.openshift.io/scc.podSecurityLabelSync": "false",
	})
        for _, v := range visitors {
             v(m)
        }
}
```

This will require modification to the other [factory](https://github.com/openshift/cluster-logging-operator/blob/master/internal/runtime/core.go) classes to take a range of visitors but it does not require changes necessarily to all the callers.

As a follow-up, we do not need to:
* we should be able move the parameters added to the reconcile method signiture to the collector factory struct and write a single "visitor" to add labels.  

Thoughts?

Reply:
@jcantrill Your proposal seems to make sense and looks more consistent, but I have a question about it. If I can set the `instance name` and `kind` in the `factory struct`, it seems reasonable because those values are immutable. However, I'm unsure about how to pass the component name to the visitor. It seems like this could potentially disrupt the idea of having a single visitor. Could you please clarify this point for me?

Reply:
I've been trying to use the [examples](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels) with our usecase and I'm not exactly certain how this maps or what is useful.  I believe we have it mostly correct in our design but should it be for a CLF instance:
* name: vector|fluentd
* instance: clf.name
* version: clo version
* component: collector
* part-of: cluster-logging
* managed-by: cluster-logging-operator

I believe you could create the visitor when creating the factory and set the visitor as a field on the factory.

Reply:
@jcantrill I've committed changes according to your comments, check please